utils/schemed-kafka-components-utils/src/main/scala/pl/touk/nussknacker/engine/schemedkafka/encode/JsonSchemaOutputValidator.scala

```diff
@@ -32,22 +41,32 @@ class JsonSchemaOutputValidator(validationMode: ValidationMode) extends LazyLogg
   /**
    * To see what's we currently supporting see SwaggerBasedJsonSchemaTypeDefinitionExtractor as well
    */
-  def validateTypingResultToSchema(typingResult: TypingResult, parentSchema: Schema)(implicit nodeId: NodeId): ValidatedNel[OutputValidatorError, Unit] =
-    validateTypingResult(typingResult, parentSchema, None)
+  def validateTypingResultAgainstSchema(typingResult: TypingResult, schema: Schema): ValidatedNel[OutputValidatorError, Unit] =
+    validateTypingResult(typingResult, schema, None)
 
   //todo: add support for: unions, enums, nested types, logical types
   final private def validateTypingResult(typingResult: TypingResult, schema: Schema, path: Option[String]): ValidatedNel[OutputValidatorError, Unit] = {
     (typingResult, schema) match {
-      case (Unknown, _) if validationMode == ValidationMode.lax =>
-        valid
+      case (Unknown, _) if validationMode == ValidationMode.lax => valid
+      case (Unknown, _) if validationMode == ValidationMode.strict => invalid(typingResult, schema, path)
       case (union: TypedUnion, _) =>
         validateUnionInput(union, schema, path)
-      case (typingResult: TypedObjectTypingResult, s: ObjectSchema) =>
-        validateRecordSchema(typingResult, s, path)
-      case (_, _) => canBeSubclassOf(typingResult, schema, path)
+      case (typingResult: TypedObjectTypingResult, s: ObjectSchema) if s.representsMap => validateMapSchema(path, s, typingResult.fields.toList: _*)
+      case (tc: TypedClass, s: ObjectSchema) if s.representsMap && tc.representsMapWithStringKeys => validateMapSchema(path, s, ("field", tc.params.tail.head))
+      case (typingResult: TypedObjectTypingResult, s: ObjectSchema) if !s.representsMap => validateRecordSchema(typingResult, s, path)
+      case (t, _) if t != Unknown || validationMode == ValidationMode.lax => canBeSubclassOf(typingResult, schema, path)
```

Comment:
isn't it `case (_, _)` ? Or are we just want to make sure we handled those cases above?